Tests/SonarScanner.MSBuild.PreProcessor.Test/Roslyn/PluginTests.cs

```diff
@@ -29,66 +28,64 @@ namespace SonarScanner.MSBuild.PreProcessor.Test.Roslyn;
 [TestClass]
 public class PluginTests
 {
-    [TestMethod]
-    public void Ctor_WhenKeyIsInvalid_Throws()
-    {
-        // 1. Null
-        Action act = () => new Plugin(null, "v1", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("key");
-
-        // 2. Empty
-        act = () => new Plugin(string.Empty, "v1", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("key");
-
-        // 3. Whitespace
-        act = () => new Plugin("\r ", "v1", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("key");
-    }
-
-    [TestMethod]
-    public void Ctor_WhenVersionIsInvalid_Throws()
+    [DataTestMethod]
+    [DataRow("pluginKey", "42", "test.zip", true)]
+    [DataRow("pluginKey", "42", null, false)]
+    [DataRow("pluginKey", null, "test.zip", false)]
+    [DataRow(null, "42", "test.zip", false)]
+    [DataRow("pluginKey", null, null, false)]
+    [DataRow(null, "42", null, false)]
+    [DataRow(null, null, "test.zip", false)]
+    [DataRow(null, null, null, false)]
+    public void Plugin_IsValid(string key, string version, string resourceName, bool isValid)
```

Comment:
Splitting this into two tests might make it easier to follow (less parameters, less brain overload)